applications/blackhole/src/bh_context.erl

```diff
@@ -197,16 +170,38 @@ set_account_id(#bh_context{}=Context, AcctId) ->
 %% @end
 %%--------------------------------------------------------------------
 -spec bindings(context()) -> ne_binaries().
-bindings(#bh_context{bindings=Bindings}) ->
-    Bindings.
+bindings(#bh_context{bindings=Bds}) ->
+    Bds.
+
+-spec bindings_from_json(wh_json:object()) -> ne_binary() | ne_binaries().
+bindings_from_json(JObj) ->
+    wh_json:get_first_defined([<<"binding">>, <<"bindings">>], JObj).
```

Comment:
This function is named `bindings_from_json` and yet can return a single binding. Why not check `binding` first and wrap it in a list; otherwise check `bindings`, returning the empty list as a default (as is, later usage will crash if neither key is included).